underscore.array.builders.js

```diff
@@ -196,6 +197,14 @@
           }));
         },[]);
       },_.map(arguments[0],function(i){return [i];}));
+    },
+    
+    // Inserts an item in an array at the specific index mutating the original
+    // array and returning it.
+    insert: function(array, index, item){
+      if (!_.isArray(array)) throw new TypeError('Expected an array as the first argument');
```

Comment:
I think this check is too restrictive, because `splice` also works with `arguments` and similar array-like objects. I think you can just remove this line.